smithy-trait-codegen/src/main/java/software/amazon/smithy/traitcodegen/TraitCodegenUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.traitcodegen;
+
+import java.net.URL;
+import java.util.List;
+import software.amazon.smithy.codegen.core.ReservedWords;
+import software.amazon.smithy.codegen.core.ReservedWordsBuilder;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.traits.TraitDefinition;
+import software.amazon.smithy.utils.CaseUtils;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.SmithyInternalApi;
+import software.amazon.smithy.utils.StringUtils;
+
+/**
+ * Provides various utility methods for trait code generation.
+ */
+@SmithyInternalApi
+public final class TraitCodegenUtils {
+    public static final Symbol JAVA_STRING_SYMBOL = TraitCodegenUtils.fromClass(String.class);
+    private static final URL RESERVED_WORDS_FILE = TraitCodegenUtils.class.getResource("reserved-words.txt");
+    private static final String SHAPE_ESCAPE_SUFFIX = "Shape";
+    private static final String MEMBER_ESCAPE_SUFFIX = "Member";
+    public static final ReservedWords SHAPE_ESCAPER = new ReservedWordsBuilder()
+            .loadCaseInsensitiveWords(RESERVED_WORDS_FILE, word -> word + SHAPE_ESCAPE_SUFFIX)
+            .build();
+    public static final ReservedWords MEMBER_ESCAPER = new ReservedWordsBuilder()
+            .loadCaseInsensitiveWords(RESERVED_WORDS_FILE, word -> word + MEMBER_ESCAPE_SUFFIX)
+            .build();
+    private static final List<String> DELIMITERS = ListUtils.of("_", " ", "-");
```

Comment:
In general this would be true, but the private fields are used in the public static ones so re-ordering in that way would result in an illegal forward reference.